spec/services/google/search_service_spec.rb

```diff
@@ -0,0 +1,55 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+require 'support/http_client_helpers'
+
+RSpec.describe Google::SearchService, type: :service do
+  describe '#search!' do
+    context 'given a successful request' do
+      it 'returns the result page html when the request succeeds' do
+        html = '<html><title>Search results</title></html>'
+        HttpClientHelpers.stub_http_adapter do |adapter, connection|
+          allow(described_class).to receive(:http_client).and_return(connection)
+          adapter.get('/search') { [200, { 'Content-Type': 'text/html' }, html] }
+        end
+
+        expect(described_class.search!('hello world')).to eq(html)
+      end
+    end
+
+    context 'given a failed request' do
```

Comment:
```
context 'given a 2xx response`
  it 'xxx

context 'given a 4xx response`
  it 'raises xxx

context 'given a 5xx response`
  it 'raises xxx

context 'given a connection fail issue`
  it 'raises xxx
```

Reply:
Fixed in 9959a387e2ca8b88f4099a56dc6bd1cd15ee4829